lib/models/partner.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const SchemaOptions = require('../options');
+
+const Partner = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+    unique: true,
+    index: true
+  },
+  /*
+   * Percentage is in decimal form, range from 0.0 to 1.0
+   */
+  revSharePercentage: {
+    type: Number,
+    min: 0,
+    max: 1,
+    required: true
```

Comment:
Might not want this to be required as it's possible future deals will used a different metric (e.g. $ per gb downloaded).